### PR TITLE
Added properties of Heyting Commutative Ring

### PR DESCRIPTION
**Pull request by guilhermehas**
Some of these properties are:

  ```agda
  leftInv→rightInv : LeftInvertible _≈_ 1# _*_ (x - y) → RightInvertible _≈_ 1# _*_ (x - y)
  rightInv→leftInv : RightInvertible _≈_ 1# _*_ (x - y) → LeftInvertible _≈_ 1# _*_ (x - y)
  leftInv→Inv : LeftInvertible _≈_ 1# _*_ (x - y) → Invertible _≈_ 1# _*_ (x - y)
  rightInv→Inv : RightInvertible _≈_ 1# _*_ (x - y) → Invertible _≈_ 1# _*_ (x - y)
  leftInvertible⇒# : LeftInvertible _≈_ 1# _*_ (x - y) → x # y
  rightInvertible⇒# : RightInvertible _≈_ 1# _*_ (x - y) → x # y
  x#0y#0→xy#0 : x # 0# → y # 0# → x * y # 0#
  #-sym : Symmetric _#_
  #-congʳ : x ≈ y → x # z → y # z
  #-congˡ : y ≈ z → x # y → x # z
  ```

### CODE REVIEW

**Comment by jamesmckinna** [comment]
The first 6 lemmas above seem to have nothing to do with `*Ring`, but rather to the multiplicative `(1#, _*_)` monoid substructure; so library SOP dictates you should move them to `Algebra.Properties.(Commutative)Monoid` and then re-import them to `HeytingCommutativeRing`?

**Comment by guilhermehas** [comment]
> The first 6 lemmas above seem to have nothing to do with `*Ring`, but rather to the multiplicative `(1#, _*_)` monoid substructure; so library SOP dictates you should move them to `Algebra.Properties.(Commutative)Monoid` and then re-import them to `HeytingCommutativeRing`?

I put the first 4 definitions in Commutative Monoid Properties. However `leftInvertible⇒#` needs `_#_` that is only defined in Heyting Commutative Ring.

File: src/Algebra/Properties/CommutativeMonoid.agda [hunk 1]
```diff
@@ -23,3 +25,18 @@ open CommutativeMonoid M
   ; assoc     to +-assoc
   ; comm      to +-comm
   )
+
+private variable
+  x : Carrier
+
+leftInv→rightInv : LeftInvertible _≈_ 0# _+_ x → RightInvertible _≈_ 0# _+_ x
```
Comment (MatthewDaggitt) [inline src/Algebra/Properties/CommutativeMonoid.agda:32]: These names should be of the form `invertibleˡ⇒invertibleʳ` to be consistent with other library naming schemes.

File: src/Algebra/Apartness/Properties/HeytingCommutativeRing.agda [hunk 2]
```diff
@@ -11,23 +11,123 @@ open import Algebra.Apartness.Bundles using (HeytingCommutativeRing)
...
+  x⁻¹ = InvX .proj₁
+
+  x⁻¹*x≈1 : x⁻¹ * (x - 0#) ≈ 1#
+  x⁻¹*x≈1 = InvX .proj₂ .proj₁
+
+  x*x⁻¹≈1 : (x - 0#) * x⁻¹ ≈ 1#
+  x*x⁻¹≈1 = InvX .proj₂ .proj₂
+
+  InvY : Invertible _≈_ 1# _*_ (y - 0#)
+  InvY = #⇒invertible y#0
```
Comment (MatthewDaggitt) [inline src/Algebra/Apartness/Properties/HeytingCommutativeRing.agda:62]: I feel like a pattern matching here (and elsewhere) would be better e.g. `(y⁻¹ , y⁻¹*y≈1 , y*y⁻¹≈1) = #⇒invertible y#0` would save a lot of lines?

File: src/Algebra/Apartness/Properties/HeytingCommutativeRing.agda [hunk 3]
```diff
@@ -11,23 +11,123 @@ open import Algebra.Apartness.Bundles using (HeytingCommutativeRing)
...
+open import Algebra.Properties.Ring ring
+  using (-0#≈0#; -‿distribˡ-*; -‿distribʳ-*; -‿anti-homo-+; -‿involutive)
+open import Relation.Binary.Definitions using (Symmetric)
+import Relation.Binary.Reasoning.Setoid as ReasonSetoid
+open import Algebra.Properties.CommutativeMonoid *-commutativeMonoid
 
+private variable
+  x y z : Carrier
+
+leftInvertible⇒# : LeftInvertible _≈_ 1# _*_ (x - y) → x # y
```
Comment (MatthewDaggitt) [inline src/Algebra/Apartness/Properties/HeytingCommutativeRing.agda:30]: Again `invertibleˡ⇒#` would be a more consistent name.

**Pull request merged**